source/Octopus.Client/OctopusAsyncClient.cs

```diff
@@ -150,110 +144,39 @@ private static async Task<IOctopusAsyncClient> Create(OctopusServerEndpoint serv
             }
         }
 
-        private static async Task<SpaceRootResource> LoadSpaceRootResource(OctopusAsyncClient client, RootResource rootResource)
-        {
-            if (client.clientOptions.SpaceContext == null)
-            {
-                var defaultSpace = await GetDefaultSpace(client, rootResource);
-                client.clientOptions.SpaceContext = defaultSpace != null ? SpaceContext.SpecificSpaceAndSystem(defaultSpace.Id) : SpaceContext.SystemOnly();
-            }
-
-            var spaceId = GetSpaceId(client.SpaceContext);
-            return !string.IsNullOrEmpty(spaceId) ?
-                await client.Get<SpaceRootResource>(rootResource.Link("SpaceHome"), new { spaceId }).ConfigureAwait(false)
-                : null;
-        }
-
-
-        /// <summary>
-        /// Gets a document that identifies the Octopus server (from /api) and provides links to the resources available on the
-        /// server. Instead of hardcoding paths,
-        /// clients should use these link properties to traverse the resources on the server. This document is lazily loaded so
-        /// that it is only requested once for
-        /// the current <see cref="IOctopusAsyncClient" />.
-        /// </summary>
-        public RootResource RootDocument => RootDocuments.RootResource;
-
-        public SpaceRootResource SpaceRootDocument => RootDocuments.SpaceRootResource;
-
-        RootResources RootDocuments { get; set; }
-
-        class RootResources
-        {
-            public RootResources(RootResource rootResource, SpaceRootResource spaceRootResource)
-            {
-                RootResource = rootResource;
-                SpaceRootResource = spaceRootResource;
-            }
-
-            public RootResource RootResource { get; }
-            public SpaceRootResource SpaceRootResource { get; }
-        }
+        [Obsolete("This property is deprecated, please use the one from Repository instead")]
+        public RootResource RootDocument => Repository?.RootDocument;
```

Comment:
I think this delegation should be the other way around - we store the root document on the client (deprecated), and the repository delegates to the root document on the client.

This way, if people are `new`ing up the client and repository manually (rather than doing `client.Repository`), then they can still access the root document on either the client or the repository. For those people that are `new`ing it up manually in existing code, they won't be doing `client.Repository = repository`